File: gurls/test/testall.m
if ~ischar(datadir)
    error('datadir is not a string')
end
if isempty(datadir)
    error('datadir is empty')
end

cd(datadir)

%% default options
fid = fopen('singlelambda.txt','w');
fprintf(fid,'median');
fclose(fid);
dlmwrite('nlambda.txt',10);
dlmwrite('nsigma.txt',5);
dlmwrite('nholdouts.txt',1);
dlmwrite('hoproportion.txt',0.2);
dlmwrite('smallnumber.txt',1e-8,'precision','%1.11e');
fid = fopen('hoperf.txt','w');
fprintf(fid,'macroavg');
fclose(fid);

%% split
dirname = 'splitho';
if ~exist(dirname,'dir'); mkdir(dirname); end
test_driver('split_ho',dirname)

%% paramsel A
dirname = 'paramselsiglam';
if ~exist(dirname,'dir'); mkdir(dirname); end
test_driver('paramsel_siglam',dirname)

dirname = 'paramselsiglamho';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('splitho/split*',dirname)
test_driver('paramsel_siglamho',dirname)

%% kernel
dirname = 'kernellinear';
if ~exist(dirname,'dir'); mkdir(dirname); end
test_driver('kernel_linear',dirname)

dirname = 'kernelgauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('paramselsiglam/paramsel-sigma.txt',dirname)
test_driver('kernel_rbf',dirname)

dirname = 'kernelchisquared';
if ~exist(dirname,'dir'); mkdir(dirname); end
test_driver('kernel_chisquared',dirname)


%% paramsel B
dirname = 'paramselloocvprimal';
if ~exist(dirname,'dir'); mkdir(dirname); end
test_driver('paramsel_loocvprimal',dirname)

dirname = 'paramselhoprimal';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('splitho/split*',dirname)
test_driver('paramsel_hoprimal',dirname)

dirname = 'paramselloocvdual_linear';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('kernellinear/kernel-K.txt',dirname)
copyfile('kernellinear/kernel-type.txt',dirname)
test_driver('paramsel_loocvdual',dirname)

dirname = 'paramselhodual_linear';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('splitho/split*',dirname)
copyfile('kernellinear/kernel-K.txt',dirname)
copyfile('kernellinear/kernel-type.txt',dirname)
test_driver('paramsel_hodual',dirname)

dirname = 'paramselloocvdual_gauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('kernelgauss/kernel-K.txt',dirname)
copyfile('kernelgauss/kernel-type.txt',dirname)
test_driver('paramsel_loocvdual',dirname)

dirname = 'paramselhodual_gauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('splitho/split*',dirname)
copyfile('kernelgauss/kernel-K.txt',dirname)
copyfile('kernelgauss/kernel-type.txt',dirname)
test_driver('paramsel_hodual',dirname)

dirname = 'paramselhoprimalr';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('splitho/split*',dirname)
test_driver('paramsel_hoprimalr',dirname)

dirname = 'paramselhodualr_linear';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('splitho/split*',dirname)
copyfile('kernellinear/kernel-K.txt',dirname)
copyfile('kernellinear/kernel-type.txt',dirname)
test_driver('paramsel_hodualr',dirname)

dirname = 'paramselhodualr_gauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('splitho/split*',dirname)
copyfile('kernelgauss/kernel-K.txt',dirname)
copyfile('kernelgauss/kernel-type.txt',dirname)
test_driver('paramsel_hodualr',dirname)

%% rls

dirname = 'rlsprimal';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('paramselhoprimal/paramsel-lambdas.txt',dirname)
test_driver('rls_primal',dirname)

dirname = 'rlsdual_linear';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('paramselhodual_linear/paramsel-lambdas.txt',dirname)
copyfile('kernellinear/kernel-K.txt',dirname)
copyfile('kernellinear/kernel-type.txt',dirname)
test_driver('rls_dual',dirname)

dirname = 'rlsdual_gauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('paramselhodual_gauss/paramsel-lambdas.txt',dirname)
copyfile('kernelgauss/kernel-K.txt',dirname)
copyfile('kernelgauss/kernel-type.txt',dirname)
test_driver('rls_dual',dirname)

dirname = 'rlsprimalr';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('paramselhoprimalr/paramsel-lambdas.txt',dirname)
test_driver('rls_primalr',dirname)

dirname = 'rlsdualr_linear';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('paramselhodualr_linear/paramsel-lambdas.txt',dirname)
copyfile('kernellinear/kernel-K.txt',dirname)
copyfile('kernellinear/kernel-type.txt',dirname)
test_driver('rls_dualr',dirname)

dirname = 'rlsdualr_gauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('paramselhodualr_gauss/paramsel-lambdas.txt',dirname)
copyfile('kernelgauss/kernel-K.txt',dirname)
copyfile('kernelgauss/kernel-type.txt',dirname)
test_driver('rls_dualr',dirname)

%% predkernel

dirname = 'predkernelchisquared';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('kernelchisquared/kernel-type.txt',dirname)
copyfile('rlsdual_gauss/optimizer-X.txt',dirname) %va bene anche di un altro kernel, tanto prendo solo X
test_driver('predkernel_traintest',dirname)

dirname = 'predkernel_gauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('kernelgauss/kernel-type.txt',dirname)
copyfile('rlsdual_gauss/optimizer-X.txt',dirname) %va bene anche di un altro kernel, tanto prendo solo X
copyfile('paramselsiglam/paramsel-sigma.txt',dirname)
test_driver('predkernel_traintest',dirname)

%% pred

dirname = 'predprimal';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('rlsprimal/optimizer*',dirname)
test_driver('pred_primal',dirname)

dirname = 'preddual_linear';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('rlsdual_linear/optimizer*',dirname)
copyfile('kernellinear/kernel-type.txt',dirname)
test_driver('pred_dual',dirname)

dirname = 'preddual_gauss';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('rlsdual_gauss/optimizer*',dirname)
copyfile('predkernel_gauss/predkernel-K.txt',dirname)
copyfile('kernelgauss/kernel-type.txt',dirname)
test_driver('pred_dual',dirname)

%% perf

dirname = 'perfrmse';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('predprimal/pred*',dirname)
test_driver('perf_rmse',dirname)

dirname = 'perfmacroavg';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('predprimal/pred*',dirname)
test_driver('perf_macroavg',dirname)

dirname = 'perfprecrec';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('predprimal/pred*',dirname)
test_driver('perf_precrec',dirname)


%% conf

dirname = 'confgap';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('predprimal/pred*',dirname)
test_driver('conf_gap',dirname)

dirname = 'confmaxscore';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('predprimal/pred*',dirname)
test_driver('conf_maxscore',dirname)

dirname = 'confboltzman';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('predprimal/pred*',dirname)
test_driver('conf_boltzman',dirname)

dirname = 'confboltzmangap';
if ~exist(dirname,'dir'); mkdir(dirname); end
copyfile('predprimal/pred*',dirname)
test_driver('conf_boltzmangap',dirname)

File: gurls++/test/testall.cpp
#include "gmat2d.h"
#include "options.h"
#include "optlist.h"

#include "linearkernel.h"
#include "rbfkernel.h"
#include "chisquaredkernel.h"

#include "predkerneltraintest.h"

#include "precisionrecall.h"
#include "macroavg.h"
#include "rmse.h"

#include "rlsauto.h"
#include "rlsprimal.h"
#include "rlsprimalr.h"
#include "rlsdual.h"
#include "rlsdualr.h"
#include "rlspegasos.h"

#include "loocvprimal.h"
#include "loocvdual.h"
#include "fixlambda.h"
#include "fixsiglam.h"
#include "siglam.h"
#include "siglamho.h"
#include "hoprimal.h"
#include "hodual.h"

#include "primal.h"
#include "dual.h"

#include "splitho.h"

#include "boltzman.h"
#include "boltzmangap.h"
#include "gap.h"
#include "maxscore.h"

#include <cstdlib>

#define BOOST_TEST_DYN_LINK
#define BOOST_TEST_MODULE yeast

#include <boost/test/unit_test.hpp>

#include "test.h"

using namespace gurls::test;

typedef double T;


//BOOST_AUTO_TEST_SUITE(yeast)

const path yeastDataPath( path(std::string(GURLS_DATA_DIR))/path("yeast"));


//BOOST_AUTO_TEST_CASE(TestSplitHo)
//{
//    Data data(yeastDataPath, "splitho", true);

//    data.loadDefaults();

//    Fixture<T, gurls::SplitHo<T> >fixture(yeastDataPath, "splitho", data);

//    fixture.runTask();

//    fixture.checkResults("split");
//}

BOOST_AUTO_TEST_CASE(TestParamSelSiglam)
{
    Data data(yeastDataPath, "paramselsiglam", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelSiglam<T> > fixture(yeastDataPath, "paramselsiglam", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

BOOST_AUTO_TEST_CASE(TestParamSelSiglamHo)
{
    Data data(yeastDataPath, "paramselsiglamho", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelSiglamHo<T> > fixture(yeastDataPath, "paramselsiglamho", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

BOOST_AUTO_TEST_CASE(TestKernelLinear)
{
    Data data(yeastDataPath, "kernellinear", true);

    data.loadDefaults();

    Fixture<T, gurls::KernelLinear<T> > fixture(yeastDataPath, "kernellinear", data);

    fixture.runTask();

    fixture.checkResults("kernel");
}

BOOST_AUTO_TEST_CASE(TestKernelRBF)
{
    Data data(yeastDataPath, "kernelgauss", true);

    data.loadDefaults();

    Fixture<T, gurls::KernelRBF<T> > fixture(yeastDataPath, "kernelgauss", data);

    fixture.runTask();

    fixture.checkResults("kernel");
}

BOOST_AUTO_TEST_CASE(TestKernelChisquared)
{
    Data data(yeastDataPath, "kernelchisquared", true);

    data.loadDefaults();

    Fixture<T, gurls::KernelChisquared<T> > fixture(yeastDataPath, "kernelchisquared", data);

    fixture.runTask();

    fixture.checkResults("kernel");
}

BOOST_AUTO_TEST_CASE(TestParamSelLoocvPrimal)
{
    Data data(yeastDataPath, "paramselloocvprimal", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelLoocvPrimal<T> > fixture(yeastDataPath, "paramselloocvprimal", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

BOOST_AUTO_TEST_CASE(TestParamSelHoPrimal)
{
    Data data(yeastDataPath, "paramselhoprimal", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelHoPrimal<T> > fixture(yeastDataPath, "paramselhoprimal", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

BOOST_AUTO_TEST_CASE(TestParamSelLoocvDual_linearkernel)
{
    Data data(yeastDataPath, "paramselloocvdual_linear", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelLoocvDual<T> > fixture(yeastDataPath, "paramselloocvdual_linear", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

BOOST_AUTO_TEST_CASE(TestParamSelHoDual_linearkernel)
{
    Data data(yeastDataPath, "paramselhodual_linear", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelHoDual<T> > fixture(yeastDataPath, "paramselhodual_linear", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

BOOST_AUTO_TEST_CASE(TestParamSelLoocvDual_gausskernel)
{
    Data data(yeastDataPath, "paramselloocvdual_gauss", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelLoocvDual<T> > fixture(yeastDataPath, "paramselloocvdual_gauss", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

BOOST_AUTO_TEST_CASE(TestParamSelHoDual_gausskernel)
{
    Data data(yeastDataPath, "paramselhodual_gauss", true);

    data.loadDefaults();

    Fixture<T, gurls::ParamSelHoDual<T> > fixture(yeastDataPath, "paramselhodual_gauss", data);

    fixture.runTask();

    fixture.checkResults("paramsel");
}

//BOOST_AUTO_TEST_CASE(TestParamSelHoPrimalr)
//{
//    Data data(yeastDataPath, "paramselhoprimalr", true);

//    data.loadDefaults();

//    Fixture<T, gurls::ParamSelHoPrimalr<T> > fixture(yeastDataPath, "paramselhoprimalr", data);

//    fixture.runTask();

//    fixture.checkResults("paramsel");
//}

//BOOST_AUTO_TEST_CASE(TestParamSelHoDualr_linearkernel)
//{
//    Data data(yeastDataPath, "paramselhodualr_linear", true);

//    data.loadDefaults();

//    Fixture<T, gurls::ParamSelHoDualr<T> > fixture(yeastDataPath, "paramselhodualr_linear", data);

//    fixture.runTask();

//    fixture.checkResults("paramsel");
//}

//BOOST_AUTO_TEST_CASE(TestParamSelHoDualr_gausskernel)
//{
//    Data data(yeastDataPath, "paramselhodualr_gauss", true);

//    data.loadDefaults();

//    Fixture<T, gurls::ParamSelHoDualr<T> > fixture(yeastDataPath, "paramselhodualr_gauss", data);

//    fixture.runTask();

//    fixture.checkResults("paramsel");
//}


BOOST_AUTO_TEST_CASE(TestRLSPrimal)
{
    Data data(yeastDataPath, "rlsprimal", true);

    data.loadDefaults();

    Fixture<T, gurls::RLSPrimal<T> >fixture(yeastDataPath, "rlsprimal", data);

    fixture.runTask();

    fixture.checkResults("optimizer");
}

BOOST_AUTO_TEST_CASE(TestRLSDual_linearkernel)
{
    Data data(yeastDataPath, "rlsdual_linear", true);

    data.loadDefaults();

    Fixture<T, gurls::RLSDual<T> >fixture(yeastDataPath, "rlsdual_linear", data);

    fixture.runTask();

    fixture.checkResults("optimizer");
}

BOOST_AUTO_TEST_CASE(TestRLSDual_gausskernel)
{
    Data data(yeastDataPath, "rlsdual_gauss", true);

    data.loadDefaults();

    Fixture<T, gurls::RLSDual<T> >fixture(yeastDataPath, "rlsdual_gauss", data);

    fixture.runTask();

    fixture.checkResults("optimizer");
}


//BOOST_AUTO_TEST_CASE(TestRLSPrimalr)
//{
//    Data data(yeastDataPath, "rlsprimalr", true);

//    data.loadDefaults();

//    Fixture<T, gurls::RLSPrimalr<T> >fixture(yeastDataPath, "rlsprimalr", data);

//    fixture.runTask();

//    fixture.checkResults("optimizer");
//}

//BOOST_AUTO_TEST_CASE(TestRLSDualr_linearkernel)
//{
//    Data data(yeastDataPath, "rlsdualr_linear", true);

//    data.loadDefaults();

//    Fixture<T, gurls::RLSDualr<T> >fixture(yeastDataPath, "rlsdualr_linear", data);

//    fixture.runTask();

//    fixture.checkResults("optimizer");
//}

//BOOST_AUTO_TEST_CASE(TestRLSDualr_gausskernel)
//{
//    Data data(yeastDataPath, "rlsdualr_gauss", true);

//    data.loadDefaults();

//    Fixture<T, gurls::RLSDualr<T> > fixture(yeastDataPath, "rlsdualr_gauss", data);

//    fixture.runTask();

//    fixture.checkResults("optimizer");
//}


BOOST_AUTO_TEST_CASE(TestPredKernelTrainTest_chisquaredkernel)
{
    Data data(yeastDataPath, "predkernelchisquared", true);

    data.loadDefaults();

    Fixture<T, gurls::PredKernelTrainTest<T> > fixture(yeastDataPath, "predkernelchisquared", data);

    fixture.runTask();

    fixture.checkResults("predkernel");
}

BOOST_AUTO_TEST_CASE(TestPredKernelTrainTest_gausskernel)
{
    Data data(yeastDataPath, "predkernel_gauss", true);

    data.loadDefaults();

    Fixture<T, gurls::PredKernelTrainTest<T> > fixture(yeastDataPath, "predkernel_gauss", data);

    fixture.runTask();

    fixture.checkResults("predkernel");
}

BOOST_AUTO_TEST_CASE(TestPredPrimal)
{
    Data data(yeastDataPath, "predprimal", true);

    data.loadDefaults();

    Fixture<T, gurls::PredPrimal<T> > fixture(yeastDataPath, "predprimal", data);

    fixture.runTask();

    fixture.checkResults("pred");
}

BOOST_AUTO_TEST_CASE(TestPredDual_linearkernel)
{
    Data data(yeastDataPath, "preddual_linear", true);

    data.loadDefaults();

    Fixture<T, gurls::PredDual<T> > fixture(yeastDataPath, "preddual_linear", data);

    fixture.runTask();

    fixture.checkResults("pred");
}

BOOST_AUTO_TEST_CASE(TestPredDual_gausskernel)
{
    Data data(yeastDataPath, "preddual_gauss", true);

    data.loadDefaults();

    Fixture<T, gurls::PredDual<T> > fixture(yeastDataPath, "preddual_gauss", data);

    fixture.runTask();

    fixture.checkResults("pred");
}

BOOST_AUTO_TEST_CASE(TestPerfRmse)
{
    Data data(yeastDataPath, "perfrmse", true);

    data.loadDefaults();

    Fixture<T, gurls::PerfRmse<T> >fixture(yeastDataPath, "perfrmse", data);

    fixture.runTask();

    fixture.checkResults("perf");
}

BOOST_AUTO_TEST_CASE(TestPerfMacroAvg)
{
    Data data(yeastDataPath, "perfmacroavg", true);

    data.loadDefaults();

    Fixture<T, gurls::PerfMacroAvg<T> >fixture(yeastDataPath, "perfmacroavg", data);

    fixture.runTask();

    fixture.checkResults("perf");
}

BOOST_AUTO_TEST_CASE(TestPerfPrecRec)
{
    Data data(yeastDataPath, "perfprecrec", true);

    data.loadDefaults();

    Fixture<T, gurls::PerfPrecRec<T> >fixture(yeastDataPath, "perfprecrec", data);

    fixture.runTask();

    fixture.checkResults("perf");
}

BOOST_AUTO_TEST_CASE(TestConfGap)
{
    Data data(yeastDataPath, "confgap", true);

    data.loadDefaults();

    Fixture<T, gurls::ConfGap<T> >fixture(yeastDataPath, "confgap", data);

    fixture.runTask();

    fixture.checkResults("conf");
}

BOOST_AUTO_TEST_CASE(TestConfMaxScore)
{
    Data data(yeastDataPath, "confmaxscore", true);

    data.loadDefaults();

    Fixture<T, gurls::ConfMaxScore<T> >fixture(yeastDataPath, "confmaxscore", data);

    fixture.runTask();

    fixture.checkResults("conf");
}

BOOST_AUTO_TEST_CASE(TestConfBoltzman)
{
    Data data(yeastDataPath, "confboltzman", true);

    data.loadDefaults();

    Fixture<T, gurls::ConfBoltzman<T> >fixture(yeastDataPath, "confboltzman", data);

    fixture.runTask();

    fixture.checkResults("conf");
}

BOOST_AUTO_TEST_CASE(TestConfBoltzmanGap)
{
    Data data(yeastDataPath, "confboltzmangap", true);

    data.loadDefaults();

    Fixture<T, gurls::ConfBoltzmanGap<T> >fixture(yeastDataPath, "confboltzmangap", data);

    fixture.runTask();

    fixture.checkResults("conf");
}

//BOOST_AUTO_TEST_SUITE_END()
